### ISSUE TITLE
`fromInfinite` errors

### ISSUE THREAD

**Issue by pgujjula** [Bodigrim/chimera#40]
Running the following example:
```haskell
naturals :: UChimera Int
naturals = fromInfinite (0...)

main :: IO ()
main = print (index naturals 0)
```
results in the error
```
Data.Primitive.Array.fromListN: list length greater than specified size
CallStack (from HasCallStack):
  error, called at ./Data/Primitive/Array.hs:373:19 in primitive-0.8.0.0-EBSr4O8W5US1YYODUpLUQc:Data.Primitive.Array
```
Can you confirm if you can also reproduce this @Bodigrim? I can prepare a PR for this one as well. Thanks!

**Comment by Bodigrim** [comment]
Yep, I can reproduce this one as well.

**Comment by pgujjula** [comment]
Opened PR https://github.com/Bodigrim/chimera/pull/41 with a fix

**Comment by Bodigrim** [comment]
Fixed in #41.
